framework/graphics/vulkan_resources_util.cpp

```diff
@@ -2647,5 +2647,113 @@ VkResult VulkanResourcesUtil::BlitImage(VkCommandBuffer       command_buffer,
     return VK_SUCCESS;
 }
 
+/**
+ * @brief Computes the required byte size of host memory referenced by a structure
+ *        for a copy-buffer-to-image or a copy-image-to-buffer operation.
+ *
+ * @note Origin: Adapted from Vulkan-ValidationLayers/layers/state_tracker/image_stat.cpp.
+ *       Mirrors the logic used in the Validation Layers to determine how many bytes are
+ *       consumed based on the copy region, image format, and block/texel sizing rules.
+ *
+ * @param region        The region structure describing the copy (e.g. `VkMemoryToImageCopy`).
+ * @param array_layers  The total number of array layers in the destination image (used when
+ *                      layerCount is VK_REMAINING_ARRAY_LAYERS).
+ * @param format        The VkFormat of the destination image.
+ *
+ * @return VkDeviceSize The number of bytes that must be available from region.pHostPointer
+ *                      to satisfy the copy described by 'region'.
+ *
+ * @details
+ * - Handles depth/stencil special cases per the Vulkan specification.
+ * - Accounts for block-compressed formats by converting to texel-block units and rounding up for partial blocks.
+ * - Returns 0 for invalid/empty copies; callers should already have guards for those cases.
+ */
+template <typename RegionCopy>
+VkDeviceSize GetBufferSizeFromCopyImage(const RegionCopy& region, uint32_t array_layers, VkFormat format)
```

Comment:
This looks like a static function. Let's use `static` in the definition as well.

Nitpick: There is a similar function `GetImageResourceSizesOptimal` which calculates the size required for an image but for the gpu/implementation side. The added documentation here is great and helps a lot to tell what this new function does but I think there might still be some confusion about how the other function is different. Can we do something about that?